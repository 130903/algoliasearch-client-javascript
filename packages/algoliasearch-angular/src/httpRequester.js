// @flow
// import { HttpClient, HttpHeaders } from '@angular/common/http';
// import encode from 'querystring-es3/encode';
// import type { RequesterArgs } from 'algoliasearch-requester';

// const httpClient = new HttpClient();
/*
export default function httpRequester({
  // todo: add override possibility
  // abortController,
  body,
  method,
  url,
  requestOptions,
  timeout: originalTimeout,
  requestType,
}: RequesterArgs) {
  let headers = new HttpHeaders();

  headers = headers.set(
    'content-type',
    method === 'POST' ? 'application/x-www-form-urlencoded' : 'application/json'
  );

  headers = headers.set('accept', 'application/json');

  const url = rawUrl + (rawUrl.includes('?') ? '&' : '?') + encode(headers);

  const transferStateKey = makeStateKey(`ngais(${opts.body})`);

  if (transferState.hasKey(transferStateKey)) {
    const resp = JSON.parse(transferState.get(transferStateKey, {}));
    return Promise.resolve({
      statusCode: resp.status,
      body: resp.body,
      headers: resp.headers,
    });
  }

  return new Promise((resolve, reject) => {
    httpClient
      .request(opts.method, url, {
        headers,
        body: opts.body,
        observe: 'response',
      })
      .subscribe(
        resp => {
          transferState.set(transferStateKey, JSON.stringify(resp));
          resolve({
            statusCode: resp.status,
            body: resp.body,
            headers: resp.headers,
          });
        },
        resp =>
          reject({
            statusCode: resp.status,
            body: resp.body,
            headers: resp.headers,
          })
      );
  });
}
*/
