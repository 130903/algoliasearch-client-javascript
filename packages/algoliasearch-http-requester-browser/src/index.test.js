import httpRequester from './';
import fetchMock from 'jest-fetch-mock';

export const snapshotAll = requests =>
  requests.map(req => expect(req).resolves.toMatchSnapshot());

beforeAll(() => {
  global.fetch = fetchMock;
});

it('most simple search', () => {
  const requests = [
    httpRequester({
      body: {
        requests: {
          query: 'hello world',
        },
      },
      connectTimeout: 1000,
      method: 'POST',
      requestOptions: undefined,
      timeout: 2000,
      url: {
        hostname: 'some_app-dsn.algolia.net',
        pathname: '/1/indexes/*/queriesundefined',
      },
    }),
  ];

  snapshotAll(requests);
});
