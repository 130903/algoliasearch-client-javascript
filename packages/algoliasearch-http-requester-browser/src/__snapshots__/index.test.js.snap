// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`most simple search 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": true,
  "size": 0,
  "status": 200,
  "statusText": "OK",
  "timeout": 0,
  "url": undefined,
}
`;
